src/vs/editor/contrib/hover/browser/markdownHoverParticipant.ts

```diff
@@ -125,16 +170,148 @@ export class MarkdownHoverParticipant implements IEditorHoverParticipant<Markdow
+			|| this._focusMetadata.element === undefined
+			|| !this._anchor
+			|| !this._context
+			|| !this._context.disposables
+			|| !this._verbosityLevels
+		) {
+			return;
+		}
+		const currentVerbosityLevel = this._verbosityLevels.get(this._focusMetadata.index);
+		const provider = this._providers[this._focusMetadata.index];
+		const model = this._editor.getModel();
+		if (!provider || !model || currentVerbosityLevel === undefined) {
+			return;
+		}
+		const verbosityLevel = currentVerbosityLevel + (increaseVerbosity ? 1 : -1);
+		this._verbosityLevels.set(this._focusMetadata.index, verbosityLevel);
+		const position = new Position(this._anchor.range.startLineNumber, this._anchor.range.startColumn);
+		const context = { verbosityLevel };
+		let hover: Hover | null | undefined;
+		try {
+			hover = await Promise.resolve(provider.provideHover(model, position, CancellationToken.None, context));
+		} catch (e) {
+			onUnexpectedExternalError(e);
+		}
+		if (!hover) {
+			return;
+		}
+		const renderedMarkdown = this._renderMarkdownHoversAndActions(
+			hover.contents,
+			this._focusMetadata.index,
+			hover.verbosityMetadata,
+			this._context.disposables
+		);
+		this._focusMetadata.focusRemains = true;
+		this._focusMetadata.element.replaceWith(renderedMarkdown);
+		this._focusMetadata.element = renderedMarkdown;
+		this._context.onContentsChanged();
+		renderedMarkdown.focus();
+	}
+
+	private _renderMarkdownHoversAndActions(
+		hoverContents: IMarkdownString[],
+		hoverIndex: number,
+		verbosityMetadata: HoverVerbosityMetadata | undefined,
+		store: DisposableStore,
+	): HTMLElement {
+
+		const contentsWrapper = $('div.hover-row');
+		contentsWrapper.tabIndex = 0;
+		const contents = $('div.hover-row-contents');
+		contentsWrapper.appendChild(contents);
+		renderMarkdownInContainer(
+			this._editor,
+			this._context,
+			contents,
+			hoverContents,
+			this._languageService,
+			this._openerService,
+			store
+		);
+		if (!verbosityMetadata || !this._context || !this._context.disposables || !this._verbosityLevels) {
+			return contentsWrapper;
+		}
+		const actionsContainer = $('div.verbosity-actions');
+		contentsWrapper.appendChild(actionsContainer);
+
+		this._renderHoverExpansionAction(actionsContainer, { increase: true, activated: verbosityMetadata.canIncreaseVerbosity ?? false }, store);
+		this._renderHoverExpansionAction(actionsContainer, { increase: false, activated: verbosityMetadata.canDecreaseVerbosity ?? false }, store);
+
+		const focusTracker = this._context.disposables.add(dom.trackFocus(contentsWrapper));
+		this._context.disposables.add(focusTracker.onDidFocus(() => {
+			const focusRemains = this._focusMetadata.focusRemains;
+			this._focusMetadata = {
+				index: hoverIndex,
+				element: contentsWrapper,
+				focusRemains
+			};
+		}));
+		this._context.disposables.add(focusTracker.onDidBlur(() => {
+			if (this._focusMetadata.focusRemains) {
+				this._focusMetadata.focusRemains = false;
+				return;
+			}
```

Comment:
So the reason I have this code is because for some reason when you move focus from one of the markdown hovers to the next markdown hover, the `onDidFocus` of the second hover is called before the `onDidBlur` of the first hover. So what happens therefore is that in `onDidFocus` the focused index is updated, and directly after the focused index is set to undefined by the `onDidBlur` of the previous hover. The logic in that part of the code tries to prevent this from happening with the boolean `focusRemains`.